### PR TITLE
[ADD] Streamlit Blog Post introducing the Battery Trading Benchmark

### PR DESCRIPTION
**Pull request by Jipje**

### CODE REVIEW

File: streamlit_app.py [hunk 1]
```diff
@@ -0,0 +1,605 @@
...
+        title=title
+    )
+    col1, col2, col3 = st.columns(3)
+    with col2:
+        st.pyplot(fig)
+
+else:
+    print('The solver could not find an optimal solution.')
+
+hour_of_system = max_power_kw / max_battery_capacity_kwh / 1
```
Comment (Jipje) [inline streamlit_app.py]: This doesn't work. Shows 0 in default

File: streamlit_app.py [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+
+Aside from the crucial role energy storage plays in the energy transition, energy storage can utilize the energy
+ markets to generate revenue.
+By utilizing powerful algorithms and strategies, energy storage can make trades and sell their capacity to the
+ highest bidder.
+They can earn revenue through price arbitrage for example.
+Utilizing wholesale markets to buy energy when the price is low, and sell it at a later moment when prices spike.
+Or energy storage can sell their flexible capabilities directly to system operators in so-called capacity markets.
+
+GIGA Storage is develops, realizes and operates large-scale (Battery) Energy Storage Systems or (B)ESS for short.
```
Comment (Jipje) [inline streamlit_app.py]: First reference of GIGA Storage can contain HTML link

File: blog_post/blog_post.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+from visualizer import plot_power_schedule_capacity_and_prices
+
+if __name__ == '__main__':
+    # ---------- USER INPUT ----------
+    # 1. Define the limits of your ESS
+    max_power_kw = 1000
+    max_battery_capacity_kwh = 2000
+    allowed_cycles = 1.5
+    charge_efficiency = 0.93
+    discharge_efficiency = 0.93
```
Comment (Koen98) [inline blog_post/blog_post.py]: # Depending on your ESS, charge and discharge efficiency may vary

Comment (Koen98) [reply blog_post/blog_post.py]: Ho ik wilde dit als comment in de code zetten eigenlijk

File: streamlit_app.py [hunk 4]
```diff
@@ -0,0 +1,608 @@
...
+As such, you are free to define the limits of your own BESS.
+By default, a 2 hour system with 1 MW and 87% Round-Trip Efficiency is configured.
+If you are unsure what to configure for the other parameters, feel free to skip this part, and read on.
+""")
+
+# ---------- USER INPUT ----------
+# 1. Define the limits of your BESS
+col1, col2, col3, col4, col5, col6, col7 = st.columns(7)
+with col1:
+    max_power_kw = st.number_input("BESS Power (kW)", 100, 10000, value=1000, step=100)
```
Comment (Koen98) [inline streamlit_app.py]: Ik zou graag ieder formaat BESS willen kunnen toevoegen, is het mogelijk dat er geen limieten worden gesteld op vermogen en capaciteit?

File: visualizer.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
+import datetime as dt
+
+import numpy as np
+from matplotlib import pyplot as plt
+
+from benchmark import PriceScheduleDataFrame
+
+
+def plot_power_schedule_capacity_and_prices(
+        price_schedule_df: PriceScheduleDataFrame,
```
Comment (Koen98) [inline visualizer.py]: Would it be possible to add a different price schedule of a different market, such as the intraday or imbalance market? And how could I implement these? I hope that the revenues from the imbalance market allow me to earn my battery back quicker.

File: streamlit_app.py [hunk 6]
```diff
@@ -194,3 +195,113 @@
...
+    solver=solver,
+    charge_power=charge_power,
+    discharge_power=discharge_power,
+    min_battery_capacity_kwh=min_battery_capacity_kwh,
+    max_battery_capacity_kwh=max_battery_capacity_kwh,
+    initial_battery_capacity_kwh=initial_battery_capacity_kwh,
+    final_battery_capacity_kwh=final_battery_capacity_kwh,
+    length_of_timestep_hour=length_of_timestep_hour,
+    charge_efficiency=1,  # 100% RTE for now
+    discharge_efficiency=1,
```
Comment (Koen98) [inline streamlit_app.py]: Would it be possible to add the parameter self-discharge? The self-discharge would only apply when the battery is not (dis)charging. This value would be a percentage discharge of the available SoC over time, for instance: 3%/month of self-discharge for li-ion batteries, during idling.

**Review by Koen98 — APPROVED** [review]
Looks very good, story is clear and visuals are nice. Great work.

**Review by Koen98 — APPROVED** [review]
Code works very neatly and looks good.

**Review by Koen98 — APPROVED** [review]

**Review by Koen98 — APPROVED** [review]

File: visualizer.py (hunk 5 above)
Comment (Jipje) [reply visualizer.py]: The imbalance market can be encoded into the `PriceScheduleDataFrame` structure. The combination of these different markets is not yet in the scope of this project.

File: streamlit_app.py (hunk 6 above)
Comment (Jipje) [reply streamlit_app.py]: If self-discharge acts on a month scale, I do not think it is relevant to implement ASAP into what is currently a daily optimisation model. E.g. the BESS will be so active that self-discharge (especially with li-ion) is negligble.

That being said, if this benchmark were to be updated to run easily on monthly or yearly market data, then implementing self-discharge should definitely be a consideration.

**Pull request merged**